File: ios/Packages/SolennixCore/Sources/SolennixCore/QuickQuoteTransferData.swift
import Foundation
import Observation

public struct QuickQuoteTransferData {
    public let products: [(productId: String, productName: String, quantity: Int, unitPrice: Double)]
    public let extras: [(description: String, cost: Double, price: Double, excludeUtility: Bool)]
    public let discountType: DiscountType
    public let discountValue: Double
    public let requiresInvoice: Bool
    public let numPeople: Int

    public init(
        products: [(productId: String, productName: String, quantity: Int, unitPrice: Double)],
        extras: [(description: String, cost: Double, price: Double, excludeUtility: Bool)],
        discountType: DiscountType,
        discountValue: Double,
        requiresInvoice: Bool,
        numPeople: Int
    ) {
        self.products = products
        self.extras = extras
        self.discountType = discountType
        self.discountValue = discountValue
        self.requiresInvoice = requiresInvoice
        self.numPeople = numPeople
    }
}

@Observable
public class QuickQuoteDataHolder {
    public static let shared = QuickQuoteDataHolder()
    public var pendingData: QuickQuoteTransferData?
    private init() {}
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/QuickQuoteTransferData.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class QuickQuoteTransferData(
    val products: List<QuoteTransferProduct>,
    val extras: List<QuoteTransferExtra>,
    @SerialName("discount_type") val discountType: String, // "percent" or "fixed"
    @SerialName("discount_value") val discountValue: Double,
    @SerialName("requires_invoice") val requiresInvoice: Boolean,
    @SerialName("num_people") val numPeople: Int
)

@Serializable
data class QuoteTransferProduct(
    @SerialName("product_id") val productId: String,
    @SerialName("product_name") val productName: String,
    val quantity: Int,
    @SerialName("unit_price") val unitPrice: Double
)

@Serializable
data class QuoteTransferExtra(
    val description: String,
    val cost: Double,
    val price: Double,
    @SerialName("exclude_utility") val excludeUtility: Boolean,
    @SerialName("include_in_checklist") val includeInChecklist: Boolean = true
)

object QuickQuoteDataHolder {
    var pendingData: QuickQuoteTransferData? = null
}
